plugins/packages/appwrite/lib/index.ts

```diff
@@ -1,21 +1,27 @@
 import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-plugins/common';
-import { SourceOptions, QueryOptions } from './types';
+import { SourceOptions, QueryOptions, ReturnObject } from './types';
 import sdk from 'node-appwrite';
 import { bulkUpdate, createDocument, deleteDocument, getDocument, queryCollection, updateDocument } from './operations';
 const JSON5 = require('json5');
 
 export default class Appwrite implements QueryService {
   async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
     const database = await this.getConnection(sourceOptions);
+    const { database_id } = sourceOptions;
     const operation = queryOptions.operation;
     const body = this.returnObject(queryOptions.body);
+    const isBodyEmpty = !Object.keys(body).length;
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/ToolJet/ToolJet/pull/10047#issuecomment-2161486769>#37201c</a><span> - 06/11/2024, 07:45 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'isBodyEmpty' check is performed after the 'body' is parsed. If the 'body' is not a valid JSON, it will throw an error before reaching the 'isBodyEmpty' check. This can lead to unhandled exceptions.
</div>
<div id="fix">
<b>Fix:</b> Add a try-catch block around the 'this.returnObject(queryOptions.body)' to handle JSON parsing errors gracefully.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -11,7 +11,11 @@
  const body = this.returnObject(queryOptions.body);
 +    let isBodyEmpty = false;
 +    try {
 +      isBodyEmpty = !Object.keys(body).length;
 +    } catch (error) {
 +      throw new Error('Invalid JSON body');
 +    }
  let result = {};
</code></pre>
</div>
</details>
</div>
<br>


</div>